hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +61,106 @@
     private long countMapWithWANReplication() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getWanReplicationRef() != null).count();
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static Predicate<MapConfig> isMapStoreEnabled() {
+        return p -> p.getMapStoreConfig().isEnabled();
+    }
+
+    static Predicate<MapConfig> isMapStoreDisabled() {
+        return p -> !p.getMapStoreConfig().isEnabled();
+    }
+
+    static class LatencyInfo {
+        private static final LatencyInfo ZERO = new LatencyInfo(0, 0);
+        private final long totalLatency;
+        private final long totalOperation;
+
+        LatencyInfo(long totalLatency, long totalOperation) {
+            this.totalLatency = totalLatency;
+            this.totalOperation = totalOperation;
+        }
+
+        LatencyInfo sum(LatencyInfo latencyInfo) {
+            return new LatencyInfo(this.totalLatency + latencyInfo.totalLatency, this.totalOperation + latencyInfo.totalOperation);
+        }
+    }
+
+
+    private long mapPutLatency(Node node, Predicate<MapConfig> predicate) {
+
+        LatencyInfo latencyInfo = Optional.ofNullable(mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
+                .filter(mapStats -> mapStats.getPutOperationCount() != 0L)
+                .map(mapStats -> new LatencyInfo(mapStats.getTotalPutLatency(), mapStats.getPutOperationCount()))
+                .reduce(LatencyInfo.ZERO, LatencyInfo::sum)).orElse(LatencyInfo.ZERO);
+
+        return latencyInfo.totalOperation == 0 ? -1 : (latencyInfo.totalLatency / latencyInfo.totalOperation);
```

Comment:
There are two problems with this calculation:
 - it is duplicated in the two latency methods
 - it performs the computation on the fields of an other object (that is a code smell, AFAIR it is discussed in detail in the Refactoring book written by Fowler)
 
 So please move it into a new method of the `LatencyInfo` class, that solves both problems